services/director-v2/tests/unit/test_core_settings.py

```diff
@@ -157,17 +162,57 @@ def test_expected_failure_dynamic_sidecar_settings(
 def test_services_custom_constraints(
     custom_constraints: str,
     expected: list[str],
-    project_env_devel_environment,
+    project_env_devel_environment: EnvVarsDict,
     monkeypatch: MonkeyPatch,
 ) -> None:
     monkeypatch.setenv("DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS", custom_constraints)
-    settings = AppSettings()
+    settings = AppSettings.create_from_envs()
     assert type(settings.DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS) == list
     assert settings.DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS == expected
 
 
 def test_services_custom_constraints_default_empty_list(
-    project_env_devel_environment,
+    project_env_devel_environment: EnvVarsDict,
 ) -> None:
-    settings = AppSettings()
+    settings = AppSettings.create_from_envs()
     assert settings.DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS == []
+
+
+def test_DynamicSidecarSettings_in_development(
```

Comment:
I really don't like this upper case, since I can understand what it is from the context. You are breaking a convention. But this is just me.

Reply:
My point was to stress that I am testing the class ``DynamicSidecarSettings`` 
but I will rewrite as ``test_class_dynamicsidecarsettings_in_development``